assets/app/scripts/services/data.js

```diff
@@ -641,6 +640,9 @@ angular.module('openshiftConsole')
         return;
       }
 
+      // Reset the retries count once we've received a message with no error.
+      this._watchWebsocketRetries(type, context, 0);
```

Comment:
hmm... it just occurred to me that if we never get a message, we'll never reset this as successful... after 5 timeouts, we'll show a connection interrupted error